crates/solvers/src/domain/auction.rs

```diff
@@ -15,6 +15,13 @@ pub struct Auction {
     pub deadline: chrono::DateTime<chrono::Utc>,
 }
 
+impl Auction {
+    /// Returns the [`Price`] of a token.
+    pub fn token_price(&self, token: &eth::TokenAddress) -> Option<Price> {
+        self.tokens.get(token)?.reference_price
+    }
+}
```

Comment:
Just a reminder that now after https://github.com/cowprotocol/services/pull/1513 you should just use `auction.tokens.reference_price`.